pandas/tests/plotting/test_misc.py

```diff
@@ -433,3 +433,98 @@ def test_dictionary_color(self):
         ax = df1.plot(kind="line", color=dic_color)
         colors = [rect.get_color() for rect in ax.get_lines()[0:2]]
         assert all(color == expected[index] for index, color in enumerate(colors))
+
+    @pytest.mark.slow
+    def test_has_externally_shared_axis(self):
+        func = plotting._matplotlib.tools._has_externally_shared_axis
+
+        # Test x-axis
+        fig = self.plt.figure()
+        plots = fig.subplots(2, 4)
+
+        # Create *externally* shared axes for first and third columns
+        plots[0][0] = self.plt.subplot(231, sharex=plots[1][0])
+        plots[0][2] = self.plt.subplot(233, sharex=plots[1][2])
```

Comment:
```suggestion
        fig = self.plt.figure()
        plots = fig.subplots(2, 4)

        # Create *externally* shared axes for first and third columns
        plots[0][0] = fig.add_subplot(231, sharex=plots[1][0])
        plots[0][2] = fig.add_subplot(233, sharex=plots[1][2])
```

Reply:
@MarcoGorelli i meant something like this, sorry, cannot do this in that whatsnew note.

Reply:
Done